actions/contract_tests/goalieruns.js

```diff
@@ -1,19 +1,49 @@
+const Joi = require("joi");
 const pkg = require("../../package.json");
 const util = require("util");
 const exec = util.promisify(require("child_process").exec);
+const { WebClient } = require("@slack/web-api");
 
-for (let arg of process.argv.slice(2)) {
-  test_set = pkg.config.goalieMappings[arg].resources;
-  for (resource_name in test_set) {
-    test = test_set[resource_name];
-    test_command = 'multi-tape "' + test + '" | tap-spec';
-    exec(test_command, function (err, stdout, stderr) {
-      if (err) {
-        // actually: send this to the right slack channel
-        console.error(err);
-        return err.code;
+web = new WebClient(process.env.SLACK_TOKEN);
+
+module.exports.runTests = async function runTests() {
+  const validator = Joi.string()
+    .regex(/^[a-zA-Z]*$/)
+    .required();
+
+  for (let arg of process.argv.slice(2)) {
+    try {
+      validated_arg = await Joi.compile(validator).validateAsync(arg);
+      test_set = pkg.config.goalieMappings[validated_arg].resources;
+      for (resource_name in test_set) {
+        test = test_set[resource_name];
+        test_command = 'multi-tape "' + test + '" | tap-spec';
+        exec(test_command, async function (err, stdout, stderr) {
+          if (err) {
+            try {
+              // send this to the right slack channel
+              const result = await web.chat.postMessage({
```

Comment:
Two bits of feedback/requests:
1) can we enclose the message in a code block rather than posting it as plain text? Or format it the way Sentry and Datadog format their error messages
2) seeing the entire stack trace in the channel seems like it might get noisy and cumbersome. are the results stored somewhere else we could link to instead of posting the full error?

Reply:
1. I can figure out how to do that!
2. The stack traces for these errors are always pretty short (same fixed length as in the example channels), so it likely won't get too noisy unless there are a TON of tests failing. What do you mean by stored somewhere else––as in, where would they be stored?

Reply:
They're in the monitoring run. 

I'm of two minds about your suggestion. Contract tests should not fail in production. When they do, it's a major crisis that should be immediately addressed. All the info you need should be right there without needing to go elsewhere.

Otoh, I can see a situation in which an existing crisis would cause the monitor to fail and you wouldn't want the monitor to swamp the channel. In that situation, you can always disable the monitor temporarily.